### PR TITLE
[rfr] Add CDBs commands

### PR DESCRIPTION
**Pull request by jamiehannaford**

### CODE REVIEW

File: commands/databasescommands/backupcommands/commands.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package backupcommands
+
+import "github.com/rackspace/rack/internal/github.com/codegangsta/cli"
+
+var commandPrefix = "databases database"
+var serviceClientType = "databases"
+
+// Get returns all the commands allowed for a `servers instance` request.
```
Comment (jrperritt) [inline commands/databasescommands/backupcommands/commands.go]: `databases database`

File: commands/databasescommands/backupcommands/create.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	"github.com/rackspace/rack/commandoptions"
+	"github.com/rackspace/rack/handler"
+	"github.com/rackspace/rack/internal/github.com/codegangsta/cli"
+	"github.com/rackspace/rack/internal/github.com/rackspace/gophercloud/rackspace/db/v1/backups"
+	"github.com/rackspace/rack/util"
+)
+
+var create = cli.Command{
+	Name:        "create",
+	Usage:       util.Usage(commandPrefix, "create", "--instance <instanceId> --name <name> --description <description>"),
```
Comment (jrperritt) [inline commands/databasescommands/backupcommands/create.go]: "--instance-id <instanceID> [--name <name> | --stdin name]"

File: commands/databasescommands/backupcommands/delete.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	"github.com/rackspace/rack/handler"
+	"github.com/rackspace/rack/internal/github.com/codegangsta/cli"
+	"github.com/rackspace/rack/internal/github.com/rackspace/gophercloud/rackspace/db/v1/backups"
+	"github.com/rackspace/rack/util"
+)
+
+// delete is a reserved word in Go.
+var remove = cli.Command{
+	Name:        "delete",
+	Usage:       util.Usage(commandPrefix, "delete", "[--id <backupId> | --stdin id]"),
```
Comment (jrperritt) [inline commands/databasescommands/backupcommands/delete.go]: backupID

File: commands/databasescommands/backupcommands/create.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	params := resource.Params.(*paramsCreate)
+
+	backup, err := backups.Create(command.Ctx.ServiceClient, params.opts).Extract()
+	if err != nil {
+		resource.Err = err
+		return
+	}
+
+	resource.Result = singleBackup(backup)
+}
```
Comment (jrperritt) [inline commands/databasescommands/backupcommands/create.go]: `StdinField`?

File: commands/databasescommands/backupcommands/create.go (hunk 2 above)
Comment (jrperritt) [inline commands/databasescommands/backupcommands/create.go]: The reason for the `--instance-id` change would be if we'd potentially allow a `--instance-name` parameter instead, where we'd lookup the ID by name. Is such a thing possible in the Databases service?

File: commands/databasescommands/backupcommands/restore.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+	"github.com/rackspace/rack/handler"
+	"github.com/rackspace/rack/internal/github.com/codegangsta/cli"
+	"github.com/rackspace/rack/internal/github.com/rackspace/gophercloud"
+	"github.com/rackspace/rack/internal/github.com/rackspace/gophercloud/rackspace/db/v1/instances"
+	"github.com/rackspace/rack/util"
+)
+
+var restore = cli.Command{
+	Name:        "restore",
+	Usage:       util.Usage(commandPrefix, "create", "[--id <id> | --stdin id] --name <instanceName> --flavor <flavorId> --size <instanceSize> --wait-for-completion"),
```
Comment (jrperritt) [inline commands/databasescommands/backupcommands/restore.go]: s/create/restore
s/flavor/flavor-id (and maybe add a flavor-name parameter?)

By convention, we only include (potentially) required parameters in the `Usage` string, so we can drop `--wait-for-completion` here.

File: commands/databasescommands/backupcommands/restore.go [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+	return nil
+}
+
+func (command *commandRestore) HandlePipe(resource *handler.Resource, item string) error {
+	resource.Params.(*paramsRestore).opts.RestorePoint = item
+	return nil
+}
+
+func (command *commandRestore) HandleSingle(resource *handler.Resource) error {
+	err := command.Ctx.CheckFlagsSet([]string{"id", "name", "flavor", "size"})
```
Comment (jrperritt) [inline commands/databasescommands/backupcommands/restore.go]: Are `name`, `flavor`, and `size` are always required? If so, we should check they're set in `HandleFlags`.

File: commands/databasescommands/backupcommands/restore.go [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+func flagsRestore() []cli.Flag {
+	return []cli.Flag{
+		cli.StringFlag{
+			Name:  "id",
+			Usage: "[optional; required if `stdin` isn't provided] The ID of the backup being restored.",
+		},
+		cli.StringFlag{
+			Name:  "stdin",
```
Comment (jrperritt) [inline commands/databasescommands/backupcommands/restore.go]: Is it legal to have multiple database instances with the same name?

Comment (jrperritt) [inline commands/databasescommands/backupcommands/restore.go]: If `stdin` is a reasonable field, we'll need a `StdinField` method

File: commands/databasescommands/backupcommands/commands.go (hunk 1 above)
Comment (jrperritt) [inline commands/databasescommands/backupcommands/commands.go]: correction: `databases backup`

File: commands/databasescommands/backupcommands/commands.go [hunk 8]
```diff
@@ -0,0 +1,15 @@
+package backupcommands
+
+import "github.com/rackspace/rack/internal/github.com/codegangsta/cli"
+
+var commandPrefix = "databases database"
```
Comment (jrperritt) [inline commands/databasescommands/backupcommands/commands.go]: `databases backup`

File: commands/databasescommands/databasecommands/commands.go [hunk 9]
```diff
@@ -0,0 +1,15 @@
+package databasecommands
+
+import "github.com/rackspace/rack/internal/github.com/codegangsta/cli"
+
+var commandPrefix = "databases database"
+var serviceClientType = "databases"
+
+// Get returns all the commands allowed for a `servers instance` request.
```
Comment (jrperritt) [inline commands/databasescommands/databasecommands/commands.go]: `databases database`

File: commands/databasescommands/databasecommands/create.go [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+	"github.com/rackspace/rack/commandoptions"
+	"github.com/rackspace/rack/handler"
+	"github.com/rackspace/rack/internal/github.com/codegangsta/cli"
+	"github.com/rackspace/rack/internal/github.com/rackspace/gophercloud/openstack/db/v1/databases"
+	"github.com/rackspace/rack/util"
+)
+
+var create = cli.Command{
+	Name:        "create",
+	Usage:       util.Usage(commandPrefix, "create", "--instance <instanceId> --name <name>"),
```
Comment (jrperritt) [inline commands/databasescommands/databasecommands/create.go]: --instance-id

File: commands/databasescommands/databasecommands/create.go [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+	Flags:       commandoptions.CommandFlags(flagsCreate, keysCreate),
+	BashComplete: func(c *cli.Context) {
+		commandoptions.CompleteFlags(commandoptions.CommandFlags(flagsCreate, keysCreate))
+	},
+}
+
+func flagsCreate() []cli.Flag {
+	return []cli.Flag{
+		cli.StringFlag{
+			Name:  "instance",
```
Comment (jrperritt) [inline commands/databasescommands/databasecommands/create.go]: instance-id

File: commands/databasescommands/databasecommands/create.go [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+			Usage: "[required] The ID of the instance that the database will be created on.",
+		},
+		cli.StringFlag{
+			Name:  "name",
+			Usage: "[optional; required if `stdin` isn't provided] The name of the database being deleted.",
```
Comment (jrperritt) [inline commands/databasescommands/databasecommands/create.go]: created

File: commands/databasescommands/databasecommands/create.go [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+		},
+		cli.StringFlag{
+			Name:  "stdin",
+			Usage: "[optional; required if `name` isn't provided] The field being piped into STDIN. Valid values are: name.",
+		},
+		cli.IntFlag{
```
Comment (jrperritt) [inline commands/databasescommands/databasecommands/create.go]: StringFlag? What would this set of rules look like?

File: commands/databasescommands/databasecommands/create.go [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+	opts := &databases.BatchCreateOpts{*params.opts}
+
+	err := databases.Create(command.Ctx.ServiceClient, params.instanceID, opts).ExtractErr()
+	if err != nil {
+		resource.Err = err
+		return
+	}
+
+	resource.Result = fmt.Sprintf("Database created")
+}
```
Comment (jrperritt) [inline commands/databasescommands/databasecommands/create.go]: `StdinField`?

**Comment by jamiehannaford** [comment]
I'm gonna go ahead and close this PR because I don't have much time to finish it.

**Pull request closed without merging**